graylog2-server/src/main/java/org/graylog/plugins/views/startpage/title/StartPageItemTitleRetriever.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (C) 2020 Graylog, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the Server Side Public License, version 1,
+ * as published by MongoDB, Inc.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * Server Side Public License for more details.
+ *
+ * You should have received a copy of the Server Side Public License
+ * along with this program. If not, see
+ * <http://www.mongodb.com/licensing/server-side-public-license>.
+ */
+package org.graylog.plugins.views.startpage.title;
+
+import org.graylog.grn.GRN;
+import org.graylog2.lookup.Catalog;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import java.util.Optional;
+import java.util.concurrent.ExecutionException;
+
+/**
+ * Retrieves titles from cache, using {@link org.graylog2.lookup.Catalog}.
+ */
+@Singleton
+public class StartPageItemTitleRetriever {
+
+    private final Catalog catalog;
+
+    @Inject
+    public StartPageItemTitleRetriever(final Catalog catalog) {
+        this.catalog = catalog;
+    }
+
+    public Optional<String> retrieveTitle(final GRN itemGrn) {
+        try {
+            final Optional<Catalog.Entry> entry = catalog.getEntry(itemGrn);
+            final Optional<String> title = entry.map(Catalog.Entry::title);
+            if (title.isPresent()) {
+                return title;
+            } else {
+                return entry.map(Catalog.Entry::id);
+            }
+        } catch (ExecutionException ex) {
+            return Optional.empty();
```

Comment:
What could be a reason for this exception that we do not want to bubble up to the caller and instead we just swallow it?

Reply:
LoadingCache:

  /**
   * Returns the value associated with {@code key} in this cache, first loading that value if
   * necessary. No observable state associated with this cache is modified until loading completes.
   *
   * (...)
   * @throws **ExecutionException if a checked exception was thrown while loading the value.** 
   * (...)
   */
  V get(K key) throws ExecutionException;

Our `load` method implementation in Catalog does not throw any checked exceptions.

Reply:
I see. In this case we should just use `LoadingCache#getUnchecked` in `Catalog#getEntry` to simplify the interface and the exception handling.